build-system/babel-plugins/babel-plugin-transform-promise-resolve/index.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const filepath = require('path');
+const {addNamed} = require('@babel/helper-module-imports');
+
+module.exports = function (babel, options = {}) {
+  const {types: t} = babel;
+  const promiseResolveMatcher = t.buildMatchMemberExpression('Promise.resolve');
+  const {importFrom = 'src/resolved-promise'} = options;
```

Comment:
Another way to solve the "transform recursion" issue would be if the transform early-exited when transforming the "importFrom" module.